src/UNL/UCBCN/ActiveRecord/Record.php

```diff
@@ -317,6 +326,39 @@ public static function getByAnyField($class, $field, $value, $whereAdd = '')
         return $record;
     }
 
+    public static function getManyByAnyField($class, $field, $value, $whereAdd = '') {
+        $records = array();
+        $record = new $class;
+
+        if (!empty($whereAdd)) {
+            $whereAdd = $whereAdd . ' AND ';
+        }
+
+        $mysqli = self::getDB();
+        $sql    = 'SELECT * FROM '
+                    . $record->getTable()
+                    . ' WHERE '
+                    . $whereAdd
+                    . $field . ' = "' . $mysqli->escape_string($value) . '"';
+        $result = $mysqli->query($sql);
+
+        if (false === $result) {
+            throw new Exception($mysqli->errno.':'.$mysqli->error, 500);
+        }
+
+        if ($result->num_rows == 0) {
+            return false;
+        }
+
+        while (!empty($row = $result->fetch_assoc())) {
+            $record = new $class;
```

Comment:
It's certainly a delicate balance between time and memory. Many active record implementations are not very time efficient because they make calls to the DB to fetch each record of a collection (as you mention). There _should_ certainly be a way to handle either case.

PHP 5.5 brought in "generators" to address some of the memory concerns and Iterator complexities. Just an FYI.